### PR TITLE
Marcelo week 6 assessment

### PR DESCRIPTION
**Pull request by celoazevedo**

### CODE REVIEW

File: interview-questions.md [hunk 1]
```diff
@@ -7,44 +7,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. As a developer, I am creating an API with a model called Animal that has_many Sightings, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be part of the Animal model or the Sightings model?
 
-  Your answer:
+  Your answer: To fix this we can use migration to add a new column to the Sightings model. I forgot the command to add the column but I will list it bellow. The foregin key will be called animal_id because we want to tie the Sightings table to the Animal table (that already have a id column) by using a the unique identifier (in this case the id) from the Animal table. The foreign key is always part of the belongs_to side, in this case Sightings belongs_to Animals, so the Sightings model will hold the foreign key.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:10]: Excellent!

File: interview-questions.md [hunk 2]
```diff
@@ -7,44 +7,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-
-  Researched answer:
-
+  Your answer: show, create, update, destroy. these routes need to have the information of the specific instance as the query parameter in the url.
 
+  Researched answer: I was wrong about create and forgot to add edit. So new list is show, edit, update and delete. 
+    - get '/animals/:id' => 'animals#show'
+    - post '/animals' => 'animals#create'
+    - get '/animals/:id/edit' => 'animals#edit'
+    - patch or put '/animals/:id' => 'animals#update'
```
Comment (WalkerTexas-coder) [inline interview-questions.md:24]: Good catch!

File: interview-questions.md [hunk 3]
```diff
@@ -7,44 +7,62 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+
+- As a user I can see a header with 'To Do List'.
+- As a user I can see a input field.
+- As a user I can see a button.
+- As a user I can see a empty list.
+- As a user I can click the button and add the data from the input to the list
+- As a user I can see the date and time of addition next to my list item
+- As a user I can see a delete button next to my list item
+- As a user I can click on the delete button and remove the respective item
+- As a user I can click on my list item and cross the item name with a line
```
Comment (WalkerTexas-coder) [inline interview-questions.md:68]: Amazing user stories!

File: rails-commenting.rb [hunk 4]
```diff
@@ -54,15 +54,15 @@ def destroy
...
       redirect_to blog_post_path(@post)
     end
   end
 
-  # ---9)
+  # ---9) private is a keyword in Ruby that gives us another level of protection. anything below private can only be called inside the scope of the class. By having the strong params listed as private, we can assure that they will only be called inside the controller class.
   private
   def blog_post_params
-    # ---10)
+    # ---10) inside the strong params definition, we are requiring the blog_post object by referencing its key, and we using the .permit method to list the info that can be saved as part of that specific instance.
```
Comment (WalkerTexas-coder) [inline rails-commenting.rb:65]: Amazing work with the rails commenting!

File: code-challenges.test.js [hunk 5]
```diff
@@ -18,43 +18,105 @@
...
+// PSEUDO
+// create a function that takes in an array for numbers
+// iterate through the array -.map()- and elevate each element (number) to the third power - might use Math.pow()
+// use .reduce() higher order function to sum all of the elements in the new array.
+
+const sumOfAllCubed = (arr) => {
+  return arr.map((item) => {
+    return Math.pow(item, 3)
+  }).reduce((a, b) => a + b, 0)
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:122]: Good work here. Solid use of reduce syntax.

File: code-challenges.test.js [hunk 6]
```diff
@@ -18,43 +18,105 @@
...
+  // i am thinking about using .map() to access each object and its property and return it as a string sentence using string interpolation (might be abble to capitalize the name here).
+  // did a lot of string manipulation here spliting (to convert it into an array) the names in order to access the first letter of each word that makes up the name. Used indexing position and the builtin method toUpperCase() to access and capitalize the first letter and concatenated it with the remaining letters of the word using the built in method .substring().
+// return an array with 3 strings (sentences).
+// The test is passing! But I agree with whoever says that this is not ease to read....
 
+const aboutEachPerson = (arr) => {
+  return arr.map((item) => {
+    return `${item.name.split(' ')[0][0].toUpperCase()}${item.name.split(' ')[0].substring(1)} ${item.name.split(' ')[1][0].toUpperCase()}${item.name.split(' ')[1].substring(1)} is a ${item.occupation}.`
+  })
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:53]: The logic here is great.

**Review by WalkerTexas-coder — APPROVED** [review]
Marcelo this is amazing work! Your rails comments are thorough and well worded. Your code logic is awesome and your pseudo code pairs with it perfectly. I think your Interview questions are solid and would work very well in an interview setting. Honestly, great work and you should be really proud of yourself!

**Comment by celoazevedo** [comment]
Thank you!

**Pull request merged**